metrics.go

```diff
@@ -74,6 +81,20 @@ func (n nullMetrics) StartWorkerRequest(string) {
 func (n nullMetrics) Shutdown() {
 }
 
+func (n nullMetrics) QueuedWorkerRequest(name string) {}
+
+func (n nullMetrics) DequeuedWorkerRequest(name string) {}
+
+func (n nullMetrics) GetWorkerQueueDepth(name string) int {
+	return 0
+}
```

Comment:
Doesn't this mean that we will never scale in case of null metrics?

Reply:
yeah, not sure if we want to do an actual implementation. `nullMetrics` isn't really supposed to be used (it just saves us from writing `if metrics != nil { metrics.RecordMetric }` in which case it would still be `0`-ish behavior).

Reply:
Oh so metrics are always collected even if disabled? Thinking about it, it would probably still be better if the behavior of the system was decoupled from the metrics implementation.